src/hooks/telemetryPrerun.ts

```diff
@@ -42,6 +42,20 @@ const hook: Hook.Prerun = async function (options: Hooks['prerun']): Promise<voi
       config: this.config,
     });
 
+    process.on('warning', (warning) => {
+      const pluginInfo = commandExecution.getPluginInfo();
+      telemetry.record({
+        eventName: warning.name,
```

Comment:
Are these a great event name? Does that mean we need to know the warning name to query it in azure? Most warnings just default to `Warning` but if someone did `process.emitWarning('message', 'name')` we would be sending events we would NEVER query for. How would we know the event type `name`?

I would rather only send up warnings we want to know about OR have the `eventName` be hardcoded to `COMMAND_WARNING` with a `name: warningName` property. (i'm not tied to those names).

Reply:
@amphro changed to `NODE_WARNING` and then added a `warningType` property to the event